src/Compilers/Core/Portable/Compilation/IImportScope.cs

```diff
@@ -0,0 +1,111 @@
+    /// </remarks>
+    public interface IImportScope
+    {
+        /// <summary>
+        /// Aliases defined at this level of the chain.  This corresponds to <c>using X = TypeOrNamespace;</c> in C# or
+        /// <c>Imports X = TypeOrNamespace</c> in Visual Basic.  This will include global aliases if present for both
+        /// languages.
+        /// </summary>
+        ImmutableArray<IAliasSymbol> Aliases { get; }
+
+        /// <summary>
+        /// Extern aliases defined at this level of the chain.  This corresponds to <c>extern alias X;</c> in C#.  It
+        /// will be empty in Visual Basic.
+        /// </summary>
+        ImmutableArray<IAliasSymbol> ExternAliases { get; }
+
+        /// <summary>
+        /// Types or namespaces imported at this level of the chain.  This corresponds to <c>using Namespace;</c> or
+        /// <c>using static Type;</c> in C#, or <c>Imports TypeOrNamespace</c> in Visual Basic.  This will include
+        /// global namespace or type imports for both languages.
+        /// </summary>
+        ImmutableArray<ImportedNamespaceOrType> Imports { get; }
+
+        /// <summary>
+        /// Xml namespaces imported at this level of the chain.  This corresponds to <c>Imports &lt;xmlns:prefix =
+        /// "name"&gt;</c> in Visual Basic.  It will be empty in C#.
+        /// </summary>
+        ImmutableArray<ImportedXmlNamespace> XmlNamespaces { get; }
+    }
+
+    /// <summary>
+    /// Represents an <see cref="INamespaceOrTypeSymbol"/> that has been imported, and the location the import was
+    /// declared at.  This corresponds to <c>using Namespace;</c> or <c>using static Type;</c> in C#, or <c>Imports
+    /// TypeOrNamespace</c> in Visual Basic.
+    /// </summary>
+    public readonly struct ImportedNamespaceOrType
+    {
+        public INamespaceOrTypeSymbol NamespaceOrType { get; }
+        public SyntaxReference? DeclaringSyntaxReference { get; }
+
+        internal ImportedNamespaceOrType(INamespaceOrTypeSymbol namespaceOrType, SyntaxReference? declaringSyntaxReference)
+        {
+            NamespaceOrType = namespaceOrType;
+            DeclaringSyntaxReference = declaringSyntaxReference;
+        }
+    }
+
+    /// <summary>
+    /// Represents an imported xml namespace name. This corresponds to <c>Imports &lt;xmlns:prefix = "name"&gt;</c> in
+    /// Visual Basic.  It does not exist for C#.
+    /// </summary>
+    public readonly struct ImportedXmlNamespace
+    {
+        public string XmlNamespace { get; }
+        public SyntaxReference? DeclaringSyntaxReference { get; }
+
+        internal ImportedXmlNamespace(string xmlNamespace, SyntaxReference? declaringSyntaxReference)
+        {
+            XmlNamespace = xmlNamespace;
+            DeclaringSyntaxReference = declaringSyntaxReference;
+        }
+    }
+
+    /// <summary>
+    /// Simple POCO implementation of the import scope, usable by both C# and VB.
+    /// </summary>
+    internal sealed record SimpleImportScope(
```

Comment:
The defining feature of records is structural equality. Unless this type needs it, we should not use a record (we've mostly avoided using records so far in the compiler for that reason).

Reply:
wfm.